### ISSUE TITLE
How to manage many different signals via one class ? 

### ISSUE THREAD

**Issue by flashultra** [peteshand/signals#8]
At the moment I use many signals in different classes and it's a little bit hard to follow all of them.
What I tried to do is to create class SignalManager  and all signals  to use this class.

At example for SignalManager,  I will have ```signalMap``` with : 
 -  key string - type of the signal 
  - value  - the signal .

When make add, I will do 

 ```signalManager.add("OPEN_FILE_BUTTON" , clickOnButton);```

 ```signalManager.remove("OPEN_FILE_BUTTON");```

How can easy do that ? 

function  ```add``` is 

 ```
function add(eventName:String, listener:Callback, numParams:Int = 0) {
    if ( numParams ==0) { 
       var signal = new  new Signal0();
       signal.add(listener);
      signalMap[eventName] = listener;
   } else   if ( numParams == 1) { 
       var signal = new  new Signal1()<T>;
       signal.add(listener);
      signalMap[eventName] = listener;
   } else   if ( numParams == 2) { 
      var signal = new  new Signal2()<T,T>;
       signal.add(listener);
      signalMap[eventName] = listener;
   } 
}```

**Comment by peteshand** [comment]
Hey there

Personally I'm not sure if I would take this approach, you're probably better off manually creating signals and registering them in a static property.

**Comment by flashultra** [comment]
Thank you for your advise . Maybe I will do that .  
Thank you again.
